pandas/core/frame.py

```diff
@@ -6741,7 +6747,23 @@ def corr(self, method='pearson', min_periods=1):
                              "'spearman', or 'kendall', '{method}' "
                              "was supplied".format(method=method))
 
-        return self._constructor(correl, index=idx, columns=cols)
+        corr_mat = self._constructor(correl, index=idx, columns=cols)
+
+        if tri is not None:
+            mask = np.tril(np.ones_like(corr_mat,
```

Comment:
Is it possible to use `correl` instead of `corr_mat` here to avoid the conversion hit of `DataFrame` to `numpy array` when `np.ones_like` is called? (I think `correl` should be a numpy array here)

Reply:
Yes, nice catch.  `correl` should already be a `numpy` array.